use-timescale/compression/compression-design.md

```diff
@@ -80,49 +80,86 @@ because it is now just pointers to the data. When data stored like this is
+columns for ordering.
+
+## Segmenting compressed data
+
+When you have compressed your data in this way, the database has no way of
+knowing which rows need to be fetched and decompressed to resolve a query. For
+example, the database can't easily determine which rows contain data from the
+past day, as the timestamp itself is in a compressed column. You don't want to
+have to decompress all the data in a chunk, or even an entire hypertable, to
+determine which rows are required. TimescaleDB automatically includes more
+information in each batch to improve query performance.
+
+For order-by columns, TimescaleDB automatically adds additional data that stores
+the minimum and maximum value of that column. This way, the query planner can
+look at the range of timestamps in the compressed column, without having to
+decompress a batch, and determine whether the data in the batch could possibly
 match the query.
 
-When you compress your hypertable, you can also choose to specify a `SEGMENT BY`
-column. This allows you to segment compressed rows by a specific column, so that
-each compressed row corresponds to a data about a single item such as, for
-example, a specific device ID. This further allows the query planner to
-determine if the row could possibly match the query without having to decompress
-the column first. For example:
+However, using the previous example, if you wanted to query the data based on
+device ID, you would have to decompress the device column to decide if that
+batch should be part of the query. When you compress your hypertable, you can
+choose to specify a `SEGMENT BY` column. This allows you to segment compressed
+rows by a specific column, so that each compressed row corresponds to a data
+about a single item such as, for example, a specific device ID. For example, you
+could use `SEGMENT_BY` to create batches like this:
 
 |Device ID|Timestamp|Status Code|Temperature|Min Timestamp|Max Timestamp|
 |-|-|-|-|-|-|
 |A|[12:00:01, 12:00:02, 12:00:03]|[0, 0, 0]|[70.11, 70.12, 70.14]|12:00:01|12:00:03|
 |B|[12:00:01, 12:00:02, 12:00:03]|[0, 0, 0]|[70.11, 70.12, 70.14]|12:00:01|12:00:03|
 
-With the data segmented in this way, a query for device A between a time
+With the data segmented this way, a query for device A during a particular time
 interval becomes quite fast. The query planner can use an index to find those
 rows for device A that contain at least some timestamps corresponding to the
-specified interval, and even a sequential scan is quite fast since evaluating
-device IDs or timestamps does not require decompression. This means the
-query executor only decompresses the timestamp and temperature columns
-corresponding to those selected rows.
+specified interval, and even a sequential scan is quite fast, since evaluating
+device IDs or timestamps does not require decompression. This means the query
+executor only decompresses the timestamp and temperature columns corresponding
+to those selected rows.
```

Comment:
```suggestion
specified interval, and even a sequential scan is quite fast, since evaluating
device IDs or timestamps does not require decompression. This means the query
executor only decompresses the timestamp and temperature columns corresponding
to those selected rows.

This, however, comes at the expense of a worse compression ratio. If you look at the example above, you now have two rows with three items in each compressed column rather than a single row with six items in the compressed rows.
```